livy/src/main/java/org/apache/zeppelin/livy/LivySparkInterpreter.java

```diff
@@ -44,4 +44,39 @@ public String getSessionKind() {
     return "spark";
   }
 
+  @Override
+  protected String extractAppId() throws LivyException {
+    return extractStatementResult(
+        interpret("sc.applicationId", null, false, false).message()
+            .get(0).getData());
+  }
+
+  @Override
+  protected String extractWebUIAddress() throws LivyException {
+    interpret(
+        "val webui=sc.getClass.getMethod(\"ui\").invoke(sc).asInstanceOf[Some[_]].get",
+        null, false, false);
+    return extractStatementResult(
+        interpret(
+            "webui.getClass.getMethod(\"appUIAddress\").invoke(webui)", null, false, false)
+            .message().get(0).getData());
+  }
```

Comment:
the code here to extract value from Spark is language dependent - so this isn't useful for Python, R and so on right?